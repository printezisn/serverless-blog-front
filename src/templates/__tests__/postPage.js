import React from "react"
import renderer from "react-test-renderer"

import PostPage from "../postPage"

jest.mock("../../components/layout", () => "Layout")

describe("Blog post page", () => {
    it("renders correctly", () => {
        const prevPagePath = "/prevPage"
        const nextPagePath = "/nextPage"
        const page = [
            {
                id: "test-id1",
                title: "test-title1",
                description: "test-description1",
            },
            {
                id: "test-id2",
                title: "test-title2",
                description: "test-description2",
            },
            {
                id: "test-id3",
                title: "test-title3",
                description: "test-description3",
            },
        ]
        const pageContext = { prevPagePath, nextPagePath, page }
        const postPage = renderer.create(<PostPage pageContext={pageContext} />)

        expect(postPage.toJSON()).toMatchSnapshot()
    })
})
