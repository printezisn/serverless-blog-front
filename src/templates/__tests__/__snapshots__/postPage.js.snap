// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog post page renders correctly 1`] = `
<Layout
  compact={true}
>
  <div
    className="box"
  >
    <article>
      <header>
        <h2
          className="title is-4"
        >
          test-title1
        </h2>
      </header>
      <section>
        <p>
          test-description1
        </p>
      </section>
      <br />
      <footer>
        <a
          className="button"
          href="/post/read/test-id1"
        >
          Read More  
          <i
            className="fas fa-arrow-right"
          />
        </a>
      </footer>
    </article>
  </div>
  <div
    className="box"
  >
    <article>
      <header>
        <h2
          className="title is-4"
        >
          test-title2
        </h2>
      </header>
      <section>
        <p>
          test-description2
        </p>
      </section>
      <br />
      <footer>
        <a
          className="button"
          href="/post/read/test-id2"
        >
          Read More  
          <i
            className="fas fa-arrow-right"
          />
        </a>
      </footer>
    </article>
  </div>
  <div
    className="box"
  >
    <article>
      <header>
        <h2
          className="title is-4"
        >
          test-title3
        </h2>
      </header>
      <section>
        <p>
          test-description3
        </p>
      </section>
      <br />
      <footer>
        <a
          className="button"
          href="/post/read/test-id3"
        >
          Read More  
          <i
            className="fas fa-arrow-right"
          />
        </a>
      </footer>
    </article>
  </div>
  <div
    className="clearfix"
  >
    <a
      className="button is-pulled-left"
      href="/prevPage"
    >
      <i
        className="fas fa-arrow-left"
      />
        Newer Posts
    </a>
    <a
      className="button is-pulled-right"
      href="/nextPage"
    >
      Older Posts   
      <i
        className="fas fa-arrow-right"
      />
    </a>
  </div>
  <div
    className="clearfix"
  />
</Layout>
`;

exports[`Blog post page renders correctly when next page is not provided 1`] = `
<Layout
  compact={true}
>
  <div
    className="box"
  >
    <article>
      <header>
        <h2
          className="title is-4"
        >
          test-title1
        </h2>
      </header>
      <section>
        <p>
          test-description1
        </p>
      </section>
      <br />
      <footer>
        <a
          className="button"
          href="/post/read/test-id1"
        >
          Read More  
          <i
            className="fas fa-arrow-right"
          />
        </a>
      </footer>
    </article>
  </div>
  <div
    className="box"
  >
    <article>
      <header>
        <h2
          className="title is-4"
        >
          test-title2
        </h2>
      </header>
      <section>
        <p>
          test-description2
        </p>
      </section>
      <br />
      <footer>
        <a
          className="button"
          href="/post/read/test-id2"
        >
          Read More  
          <i
            className="fas fa-arrow-right"
          />
        </a>
      </footer>
    </article>
  </div>
  <div
    className="box"
  >
    <article>
      <header>
        <h2
          className="title is-4"
        >
          test-title3
        </h2>
      </header>
      <section>
        <p>
          test-description3
        </p>
      </section>
      <br />
      <footer>
        <a
          className="button"
          href="/post/read/test-id3"
        >
          Read More  
          <i
            className="fas fa-arrow-right"
          />
        </a>
      </footer>
    </article>
  </div>
  <div
    className="clearfix"
  >
    <a
      className="button is-pulled-left"
      href="/prevPage"
    >
      <i
        className="fas fa-arrow-left"
      />
        Newer Posts
    </a>
    
  </div>
  <div
    className="clearfix"
  />
</Layout>
`;

exports[`Blog post page renders correctly when no next or previous page is provided 1`] = `
<Layout
  compact={true}
>
  <div
    className="box"
  >
    <article>
      <header>
        <h2
          className="title is-4"
        >
          test-title1
        </h2>
      </header>
      <section>
        <p>
          test-description1
        </p>
      </section>
      <br />
      <footer>
        <a
          className="button"
          href="/post/read/test-id1"
        >
          Read More  
          <i
            className="fas fa-arrow-right"
          />
        </a>
      </footer>
    </article>
  </div>
  <div
    className="box"
  >
    <article>
      <header>
        <h2
          className="title is-4"
        >
          test-title2
        </h2>
      </header>
      <section>
        <p>
          test-description2
        </p>
      </section>
      <br />
      <footer>
        <a
          className="button"
          href="/post/read/test-id2"
        >
          Read More  
          <i
            className="fas fa-arrow-right"
          />
        </a>
      </footer>
    </article>
  </div>
  <div
    className="box"
  >
    <article>
      <header>
        <h2
          className="title is-4"
        >
          test-title3
        </h2>
      </header>
      <section>
        <p>
          test-description3
        </p>
      </section>
      <br />
      <footer>
        <a
          className="button"
          href="/post/read/test-id3"
        >
          Read More  
          <i
            className="fas fa-arrow-right"
          />
        </a>
      </footer>
    </article>
  </div>
  <div
    className="clearfix"
  >
    
    
  </div>
  <div
    className="clearfix"
  />
</Layout>
`;

exports[`Blog post page renders correctly when previous page is not provided 1`] = `
<Layout
  compact={true}
>
  <div
    className="box"
  >
    <article>
      <header>
        <h2
          className="title is-4"
        >
          test-title1
        </h2>
      </header>
      <section>
        <p>
          test-description1
        </p>
      </section>
      <br />
      <footer>
        <a
          className="button"
          href="/post/read/test-id1"
        >
          Read More  
          <i
            className="fas fa-arrow-right"
          />
        </a>
      </footer>
    </article>
  </div>
  <div
    className="box"
  >
    <article>
      <header>
        <h2
          className="title is-4"
        >
          test-title2
        </h2>
      </header>
      <section>
        <p>
          test-description2
        </p>
      </section>
      <br />
      <footer>
        <a
          className="button"
          href="/post/read/test-id2"
        >
          Read More  
          <i
            className="fas fa-arrow-right"
          />
        </a>
      </footer>
    </article>
  </div>
  <div
    className="box"
  >
    <article>
      <header>
        <h2
          className="title is-4"
        >
          test-title3
        </h2>
      </header>
      <section>
        <p>
          test-description3
        </p>
      </section>
      <br />
      <footer>
        <a
          className="button"
          href="/post/read/test-id3"
        >
          Read More  
          <i
            className="fas fa-arrow-right"
          />
        </a>
      </footer>
    </article>
  </div>
  <div
    className="clearfix"
  >
    
    <a
      className="button is-pulled-right"
      href="/nextPage"
    >
      Older Posts   
      <i
        className="fas fa-arrow-right"
      />
    </a>
  </div>
  <div
    className="clearfix"
  />
</Layout>
`;
